src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_9r1_9r2/WorldSound.java

```diff
@@ -14,8 +14,10 @@
 
 	@Override
 	public RecyclableCollection<ClientBoundPacketData> toData(ProtocolVersion version) {
+		String soundname = LegacySound.getSoundName(id);
 		ClientBoundPacketData serializer = ClientBoundPacketData.create(ClientBoundPacket.PLAY_WORLD_CUSTOM_SOUND_ID, version);
-		StringSerializer.writeString(serializer, version, LegacySound.getSoundName(id));
+		if(soundname == "entity.player.hurt_on_fire") soundname = "entity.player.hurt";
```

Comment:
What the? A unified remapper should be used if multiple versions need sound names remapping.